cmd/acra-migrate-keys/acra-migrate-keys.go

```diff
@@ -0,0 +1,245 @@
+/*
+ * Copyright 2020, Cossack Labs Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Package main is entry point for Acra key migration utility (`acra-migrate-keys`).
+// It is used to convert between key store formats used by Acra.
+//
+// https://docs.cossacklabs.com/pages/documentation-acra/#key-management
+package main
+
+import (
+	"errors"
+	"flag"
+
+	"github.com/cossacklabs/acra/cmd"
+	keystoreV1 "github.com/cossacklabs/acra/keystore"
+	filesystemV1 "github.com/cossacklabs/acra/keystore/filesystem"
+	keystoreV2 "github.com/cossacklabs/acra/keystore/v2/keystore"
+	keystoreApiV2 "github.com/cossacklabs/acra/keystore/v2/keystore/api"
+	filesystemV2 "github.com/cossacklabs/acra/keystore/v2/keystore/filesystem"
+	"github.com/cossacklabs/acra/logging"
+	"github.com/cossacklabs/acra/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	defaultConfigPath = utils.GetConfigPathByName("acra-migrate-keys")
+	serviceName       = "acra-migrate-keys"
+
+	defaultSrcDir       = keystoreV1.DefaultKeyDirShort
+	defaultDstDirSuffix = ".migrated"
+)
+
+// CommandLineParams - command-line options.
+type CommandLineParams struct {
+	Src, Dst KeyStoreParams
+	Misc     MiscParams
+}
+
+// KeyStoreParams - key store parameters.
+type KeyStoreParams struct {
+	KeyStoreVersion string
+	KeyDir          string
+	KeyDirPublic    string
+}
+
+// MiscParams - miscellaneous parameters.
+type MiscParams struct {
+	DryRun bool
+	Force  bool
+
+	LogDebug   bool
+	LogVerbose bool
+}
+
+// OpenMode - whether key store is source or destination.
+type OpenMode int
+
+// OpenMode constant values:
+const (
+	OpenSrc OpenMode = iota
+	OpenDst
+)
+
+// RegisterCommandLineParams registers command-line options for parsing.
+func RegisterCommandLineParams() *CommandLineParams {
+	params := &CommandLineParams{}
+	// Source key store
+	flag.StringVar(&params.Src.KeyStoreVersion, "src_keystore", "v1", "key store format to use: v1 (current), v2 (experimental)")
```

Comment:
I'd not name v2 as `experimental`, because when this thing is merged to master, we'll totally forget to rename these into "old" and "new" (or similar)